### PR TITLE
ref(security): removes unused code and adds logging

### PR DESCRIPTION
**Pull request by scefali**
This PR removes an unused call of `from_auth` that would fail since `scopes` is not an argument. I also believe that the `scopes` argument to many other functions in `access.py` is not used and would like to remove it to reduce the complexity of some of the auth code. I added some log statements if `scopes` is actually passed so I can validate if this is correct. If none of these log statements show up, then it should be safe to remove the `scopes` argument and simplify the complexity of code.

### CODE REVIEW

File: src/sentry/auth/access.py [hunk 1]
```diff
@@ -271,6 +274,17 @@ class NoAccess(BaseAccess):
...
+    """
+    Returns the an Access object corresponding to the permissions associated with this request
+    and a particular organization. The user of the request could be an actual user
+    or it could be a sentry app
+    """
+    # TODO: validate scopes argument is not used and then remove
+    if scopes is not None:
+        import traceback
+
+        logger.info("from_request.non_none_scopes: %s" % traceback.format_exc())
```
Comment (EvanPurkhiser) [inline src/sentry/auth/access.py]: Might be safer to start with a metric?

Comment (scefali) [reply src/sentry/auth/access.py]: @EvanPurkhiser because of the log volume?

**Comment by dcramer** [comment]
I thought there was a reason scopes was passable to these when I made it. There's no evidence of it being used anywhere?

**Comment by dcramer** [comment]
Also you could just do branch analysis vs logging this in prod. e.g. raise an error, or trace scopes=X in code (or the function call with a second param). No reason to do this kind of thing in prod when static analysis can achieve same result imo.

**Comment by scefali** [comment]
> I thought there was a reason scopes was passable to these when I made it. There's no evidence of it being used anywhere?
@dcramer Maybe there was a reason initially but I'm unable to find any places in the code where we use it

File: src/sentry/auth/access.py (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/auth/access.py]: just to get a quick judgment, and then you can add the logs after you verify it's not being used a lot.

Though if you're already sure logging is fine.

Comment (scefali) [reply src/sentry/auth/access.py]: @EvanPurkhiser I'm 90% sure we won't ever see the log messages at all.

**Comment by scefali** [comment]
> Also you could just do branch analysis vs logging this in prod. e.g. raise an error, or trace scopes=X in code (or the function call with a second param). No reason to do this kind of thing in prod when static analysis can achieve same result imo.

@dcramer  Are you saying just test this out locally? The reason I want to do this in prod is in case there is some code path I am not considering that is using `scopes`. Don't want to rip it out and have things break considering this is very critical code path used on every request.

**Comment by dcramer** [comment]
:shrug: I would confirm that this code path isnt needed for the OAuth2 provider pieces, otherwise idk what else it would exist for

**Comment by scefali** [comment]
> 🤷 I would confirm that this code path isnt needed for the OAuth2 provider pieces, otherwise idk what else it would exist for

@dcramer I just searched `from_request`, `from_member`, and `from_user` in sentry and getsentry to see if `scopes` is passed in. Doesn't seem like it. The log messages will confirm if my understanding is correct.

**Comment by scefali** [comment]
> 🤷 I would confirm that this code path isnt needed for the OAuth2 provider pieces, otherwise idk what else it would exist for

@dcramer I just double-checked and I'm wrong...my bad. It's used in `SentryPermission`.

**Pull request closed without merging**